File: src/php/Generic/MathBase.php
<?php

namespace EntityFX\NetBenchmark\Core\Generic {

    use EntityFX\NetBenchmark\Core\Writer;
    
    abstract class MathBase  extends BenchmarkBase {
        public function __construct($writer, $printToConsole)
        {
            parent::__construct($writer, $printToConsole);
            $this->Iterrations = 200000000;
            $this->Ratio = 3.5;
        }

        public static function DoMath($i)
        {
            $rev = 1.0 / ($i + 1.0);
            return abs($i) + acos($rev) + asin($rev) + atan($rev) +
                floor($i) + exp($rev) + cos($i) + sin($i) + M_PI + sqrt($i);
        }
    }
}

File: src/lua/generic/mathBase.lua
MathBase = class(BenchmarkBase, function(a, writer, printToConsole)
    BenchmarkBase.init(a, writer, printToConsole)
    a.iterrations = 200000000
    a.ratio = 0.5
end)

function MathBase:doMath(i, li)
    local rev = 1.0 / (i + 1.0)
    return math.abs(i) * math.acos(rev) * math.asin(rev) * math.atan(rev) +
        math.floor(li) + math.exp(rev) * math.cos(i) * math.sin(i) * math.pi + math.sqrt(i)

end